sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -15,94 +15,120 @@
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpUriRequest;
 import org.apache.http.impl.client.DefaultHttpClient;
 import org.apache.http.message.BasicNameValuePair;
 
 
+
+
 public class SectionsUtilityToolServlet extends HttpServlet {
 
+	private static final String POST = "POST";
+	private static final String GET = "GET";
+	private static final String PUT = "PUT";
 	private static final long serialVersionUID = 7284813350014385613L;
 	private static Log M_log = LogFactory.getLog(SectionsUtilityToolServlet.class);
+	private String canvasToken;
+	private String canvasURL;
+
+	
+	
+	public void init() throws ServletException {
+		M_log.debug(" Servlet init(): Called");
+	}
 	
 	protected void doGet(HttpServletRequest request,HttpServletResponse response) throws IOException {
 		M_log.debug("doGet: Called");
-		HttpClient client = new DefaultHttpClient();
-		String targetURI= "https://umich.test.instructure.com";
+		canvasRestApiCall(request, response);
+	}
+	
+	protected void doPost(HttpServletRequest request,HttpServletResponse response) throws IOException {
+		M_log.debug("doPOST: Called");
+		canvasRestApiCall(request, response);
+		
+	}
+	protected void doPut(HttpServletRequest request,HttpServletResponse response) throws IOException {
+		M_log.debug("doPut: Called");
+		canvasRestApiCall(request, response);
+		
+	}
+
+	private void canvasRestApiCall(HttpServletRequest request,
+			HttpServletResponse response) throws IOException {
+		request.setCharacterEncoding("UTF-8");
+		M_log.debug("canvasRestApiCall(): called");
+		Properties appExtSecureProperties = SectionUtilityToolFilter.appExtSecurePropertiesFile;
+		if(appExtSecureProperties!=null) {
+			canvasToken = appExtSecureProperties.getProperty(SectionUtilityToolFilter.PROPERTY_CANVAS_ADMIN);
+			canvasURL = appExtSecureProperties.getProperty(SectionUtilityToolFilter.PROPERTY_CANVAS_URL);
+		}
+		else {
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			PrintWriter out = response.getWriter();
+			out.print("Problem loading property file");
+			out.flush();
+			M_log.error("Failed to load system properties(sectionsToolProps.properties) for SectionsTool");
+			return;
+		}
 		String queryString = request.getQueryString();
-		String servletPath = request.getServletPath();
 		String pathInfo = request.getPathInfo();
 		String url=null;
 		if(pathInfo!=null&&(pathInfo.contains("courses")||pathInfo.contains("terms")||pathInfo.contains("crosslist"))) {
 		if(queryString!=null) {
-		url= targetURI+pathInfo+"?"+queryString;
-		 //url= servletPath+"?"+queryString;
-		 }
-		else {
-			url=targetURI+pathInfo;
-			//url=servletPath;
+		url= canvasURL+pathInfo+"?"+queryString;
+		 }else {
+			url=canvasURL+pathInfo;
+		}
+		String sessionId = request.getSession().getId();
+		StringBuilder loggingApiWithSessionInfo =new StringBuilder();
+		loggingApiWithSessionInfo.append("Canvas API request with Session Id:  \"");
```

Comment:
String.format could be used to get rid of a lot of appends and tighten up the code.